Tests/SwiftLexicalLookupTest/NameLookupTests.swift

```diff
@@ -803,13 +805,45 @@ final class testNameLookup: XCTestCase {
         "4️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["1️⃣"])],
         "5️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["2️⃣"])],
         "6️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["3️⃣"])],
-        "9️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["7️⃣", "8️⃣"])],
+        "9️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["7️⃣"])],
         "0️⃣": [],
       ],
       expectedResultTypes: .all(IdentifierPatternSyntax.self)
     )
   }
 
+  func testCompositeNames() {
+    assertLexicalNameLookup(
+      source: """
+        switch X {
+        case .x(let 1️⃣a, let 2️⃣b), .y(let 3️⃣a, let 4️⃣b):
+          print(5️⃣x)
+        case .z(let 7️⃣a), .smth(let 8️⃣a):
+          print(9️⃣x)
```

Comment:
Shouldn’t this be `print(a)`? `x` is never defined in this test case.

Reply:
I purposefully called it `x` to further indicate that we are not performing name matching in this case. We use `useNilAsTheParameter: true` for that. This test is supposed to check if we properly partition names equivalent names (like in the first case here).

Reply:
Oh, I missed that and thought that we were looking up `x`, not that we were collecting all names available at that location. The test makes sense now 😄